Stratis.SmartContracts.Tests/UInt256Tests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using Xunit;
+
+namespace Stratis.SmartContracts.Tests
+{
+    public class UInt256Tests
+    {
+        [Fact]
+        [Trait("UnitTest", "UnitTest")]
+        public void CanConvertToFromUlong()
+        {
+            UInt256 x = 50;
+            Assert.Equal((ulong)50, (ulong)x);
+        }
+
+        [Fact]
+        [Trait("UnitTest", "UnitTest")]
+        public void Uint256FromTooLargeUlongThrowsError()
+        {
+            UInt256 x = UInt256.Parse("010000000000000000");
+            Assert.Throws(typeof(OverflowException), () => (ulong)x);
```

Comment:
Not sure how we could simulate checked behavior. Is it really necessary to suppress these errors?

The internal `BigInteger` can handle any size number, so we can't rely on it for what you are describing.

Reply:
These details should to be documented that these types will reject overflows cases with Exception whether it is in checked state or unchecked state

Reply:
There are pros/cons of this behavior

pros 
-BigInteger is already .net built-in type and designed to handle overflow cases and we are not changing default behavior. Surely this is confusing bit of.
-It is actually safer that these numeric types doesn't let overflows. 


cons
-Checking overflow and throw exception is extra computation and gas cost and It may not needed for every case.
-It is more expected behavior that arithmetic operations behave same as other numeric types in .net regarding to checked and unchecked contexts
-Coders should to be aware these behaviors by reading documentation. 

@mrtpain Let me know your feedbacks. 

I request to you to add documentation on methods and operators for(Add,Multiply and Substract) that they will throws OverflowException and developers can easily see it.

Reply:
I don't know the differences in gas costs of using checked() or not, so my opinion isn't really backed by supporting arguments. I don't think throwing on overflow is problematic, probably actually preferred to prevent accidental overflows but again, I'm not sure the cost in gas in doing so. 

Overall, if I had to choose, I would prefer that an OverflowException be thrown automatically, rather than not having a way to check at all using `checked(UInt256Value);` or something similar.